testing/stubs.go

```diff
@@ -141,6 +141,115 @@ func TestFailOnceActivityInterceptor() activity.ActivityInterceptor {
 	}
 }
 
+func TestThrotteSignalsOnceInterceptor() fsm.DecisionInterceptor {
+	throttledSignals := make(map[string]*fsm.SignalInfo)
+	return &fsm.FuncInterceptor{
+		BeforeDecisionFn: func(decision *swf.PollForDecisionTaskOutput, ctx *fsm.FSMContext, outcome *fsm.Outcome) {
+			for _, h := range decision.Events {
+				if *h.EventType == swf.EventTypeSignalExternalWorkflowExecutionFailed {
+					if info, found := throttledSignals[*h.SignalExternalWorkflowExecutionFailedEventAttributes.WorkflowId]; found {
+						h.SignalExternalWorkflowExecutionFailedEventAttributes.WorkflowId = &info.WorkflowId
+						//rewrite workflow not found to throttling.
```

Comment:
should you check that it is actually not found here? could it be something else?